dali/test/python/test_operator_nonsilence.py

```diff
@@ -0,0 +1,103 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from functools import partial
+import itertools
+import librosa
+import numpy as np
+from nvidia.dali.pipeline import Pipeline
+import nvidia.dali.ops as ops
+import nvidia.dali.types as types
+import os
+import test_utils
+
+dali_extra = test_utils.get_dali_extra_path()
+audio_files = os.path.join(dali_extra, "db", "audio")
+
+
+def trim_ref(top_db, ref, frame_length, hop_length, input_data):
+    yt, index = librosa.effects.trim(y=input_data, top_db=top_db, ref=ref,
+                                     frame_length=frame_length,
+                                     hop_length=hop_length)
+
+    # librosa's trim function calculates power with reference to center of window,
+    # while DALI uses beginning of window. Hence the subtraction below
+    begin = index[0] - frame_length / 2
+    length = index[1] - index[0]
+    return np.array([begin]), np.array([length])
+
+
+class NonsilencePipeline(Pipeline):
+    def __init__(self, batch_size, num_threads=1, exec_async=True, exec_pipelined=True):
+        super(NonsilencePipeline, self).__init__(batch_size, num_threads, 0, seed=42,
+                                                 exec_async=exec_async,
+                                                 exec_pipelined=exec_pipelined)
+        self.input = ops.FileReader(device="cpu", file_root=audio_files)
+        self.decode = ops.AudioDecoder(device="cpu", dtype=types.FLOAT, downmix=True)
+
+        self.nonsilence = None
+
+    def define_graph(self):
+        if self.nonsilence is None:
+            raise RuntimeError(
+                "Error: you need to derive from this class and define `self.nonsilence` operator")
+        read, _ = self.input()
+        audio, rate = self.decode(read)
+        begin, len = self.nonsilence(audio)
+        return begin, len
+
+
+class NonsilenceDaliPipeline(NonsilencePipeline):
+    def __init__(self, batch_size, cutoff_value, window_size, reference_power,
+                 reset_interval):
+        super(NonsilenceDaliPipeline, self).__init__(batch_size, num_threads=1)
+        self.nonsilence = ops.NonsilentRegion(cutoff_db=cutoff_value, window_length=window_size,
+                                              reference_power=reference_power,
+                                              reset_interval=reset_interval,
+                                              device="cpu")
+
+
+class NonsilenceRosaPipeline(NonsilencePipeline):
+    def __init__(self, batch_size, cutoff_value, window_size, reference_power, reset_interval):
+        super(NonsilenceRosaPipeline, self).__init__(batch_size, num_threads=1,
+                                                     exec_async=False, exec_pipelined=False)
+        hop_length = 1
+        function = partial(trim_ref, cutoff_value,
+                           np.max if reference_power == 0. else reference_power,
```

Comment:
```suggestion
                           np.max if not reference_power else reference_power,
```

Reply:
Will it work for float values?

Reply:
Ok... I thought you were using `None` to specify "not provided"

Reply:
Done